src/overlays/fbdemos/ovl_fbdemo_triforce/z_fbdemo_triforce.c

```diff
@@ -15,35 +16,135 @@ void TransitionTriforce_SetType(void* thisx, s32 type);
+        } else if (this->state == 3) {
+            this->transPos = CLAMP_MAX(this->transPos / (1.0f - this->step), 1.0f);
+        } else if (this->state == 4) {
+            this->transPos = CLAMP_MAX(this->transPos + this->step, 1.0f);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/func_80AC5280.s")
+void TransitionTriforce_SetColor(void* thisx, u32 color) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Draw.s")
+    this->color.rgba = color;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_IsDone.s")
+void TransitionTriforce_SetType(void* thisx, s32 type) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
+
+    this->fadeDirection = type;
+}
+
+void func_80AC5280(void* thisx, s32 state) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
+
+    this->state = state;
+}
+
+void TransitionTriforce_Draw(void* thisx, Gfx** gfxP) {
+    Gfx* gfx = *gfxP;
+    Mtx* modelView;
+    s32 pad[2];
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
+    f32 rotation = this->transPos * 360.0f;
+
+    modelView = this->modelView[this->frame];
+    this->frame ^= 1;
+    guScale(&modelView[0], this->transPos * 0.625f, this->transPos * 0.625f, 1.0f);
+    guRotate(&modelView[1], rotation, 0.0f, 0.0f, 1.0f);
+    guTranslate(&modelView[2], 0.0f, 0.0f, 0.0f);
+    gDPPipeSync(gfx++);
+    gSPDisplayList(gfx++, &sTriforceWipeDL);
+    gDPSetColor(gfx++, G_SETPRIMCOLOR, this->color.rgba);
+    gDPSetCombineMode(gfx++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+    gSPMatrix(gfx++, &this->projection, G_MTX_LOAD | G_MTX_PROJECTION);
+    gSPMatrix(gfx++, &modelView[0], G_MTX_LOAD);
+    gSPMatrix(gfx++, &modelView[1], G_MTX_NOPUSH | G_MTX_MODELVIEW);
+    gSPMatrix(gfx++, &modelView[2], G_MTX_NOPUSH | G_MTX_MODELVIEW);
+    gSPVertex(gfx++, &sTriforceWipeVtx, 10, 0);
+    if (!TransitionTriforce_IsDone(this)) {
+        switch (this->fadeDirection) {
+            case 1:
+                gSP2Triangles(gfx++, 0, 4, 5, 0, 4, 1, 3, 0);
+                gSP1Triangle(gfx++, 5, 3, 2, 0);
+                break;
+
+            case 2:
+                gSP2Triangles(gfx++, 3, 4, 5, 0, 0, 2, 6, 0);
+                gSP2Triangles(gfx++, 0, 6, 7, 0, 1, 0, 7, 0);
+                gSP2Triangles(gfx++, 1, 7, 8, 0, 1, 8, 9, 0);
+                gSP2Triangles(gfx++, 1, 9, 2, 0, 2, 9, 6, 0);
+                break;
+        }
+
+    } else {
+        switch (this->fadeDirection) {
+            case 1:
+                break;
+
+            case 2:
+                gSP1Quadrangle(gfx++, 6, 7, 8, 9, 0);
+                break;
+        }
+    }
+    gDPPipeSync(gfx++);
+    *gfxP = gfx;
+}
+
+s32 TransitionTriforce_IsDone(void* thisx) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
+
+    s32 ret = 0;
```

Comment:
```suggestion
    s32 ret = false;
```
Is this temp even needed or could you `return 0;`?